scripts/merge-configs.js

```diff
@@ -2,9 +2,9 @@
 
 'use strict';
 
-var merge = require('lodash/merge');
+const merge = require('lodash/merge');
 
-hexo.on('generateBefore', function() {
+hexo.on('generateBefore', () => {
```

Comment:
```bash
FATAL Something's wrong. Maybe you can find the solution here: https://hexo.io/docs/troubleshooting.html      
TypeError: Cannot read property 'config' of undefined
    at Hexo.<anonymous> (C:\Users\mkb\1v9\themes\next\scripts\merge-configs.js:29:23)
    at Hexo.emit (events.js:205:15)
    at Hexo._generate (C:\Users\mkb\1v9\node_modules\hexo\lib\hexo\index.js:399:8)
    at C:\Users\mkb\1v9\node_modules\hexo\lib\hexo\index.js:249:22
    at tryCatcher (C:\Users\mkb\1v9\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise.js:517:31)
    at Promise._settlePromise (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise.js:574:18)
    at Promise._settlePromise0 (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise.js:619:10)
    at Promise._settlePromises (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise.js:699:18)
    at Promise._fulfill (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise.js:643:18)
    at PromiseArray._resolve (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise_array.js:126:19)      
    at PromiseArray._promiseFulfilled (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise_array.js:144:14)
    at Promise._settlePromise (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise.js:579:26)
    at Promise._settlePromise0 (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise.js:619:10)
    at Promise._settlePromises (C:\Users\mkb\1v9\node_modules\bluebird\js\release\promise.js:699:18)
    at _drainQueueStep (C:\Users\mkb\1v9\node_modules\bluebird\js\release\async.js:138:12)
    at _drainQueue (C:\Users\mkb\1v9\node_modules\bluebird\js\release\async.js:131:9)
    at Async._drainQueues (C:\Users\mkb\1v9\node_modules\bluebird\js\release\async.js:147:5)
    at Immediate.Async.drainQueues [as _onImmediate] (C:\Users\mkb\1v9\node_modules\bluebird\js\release\async.js:17:14)
    at processImmediate (internal/timers.js:439:21)
```

Reply:
On my lcoal enviroment must use `fucntion()` not `()=>`, otherwise get error.

Reply:
Tests are normal in the following environments
```
hexo: 3.9.0
hexo-cli: 2.0.0
os: Darwin 18.6.0 darwin x64
node: 12.6.0
v8: 7.5.288.22-node.14
uv: 1.30.1
zlib: 1.2.11
brotli: 1.0.7
ares: 1.15.0
modules: 72
nghttp2: 1.38.0
napi: 4
llhttp: 1.1.4
http_parser: 2.8.0
openssl: 1.1.1c
cldr: 35.1
icu: 64.2
tz: 2019a
unicode: 12.1
```

Reply:
> On my local environment must use `function()` not `()=>`, otherwise get error.

This should be a Hexo bug.